packages/autocomplete-js/src/__tests__/render.test.ts

```diff
@@ -107,8 +114,177 @@ describe('render', () => {
               },
             },
           },
+        ];
+      },
+      render({ createElement }, root) {
+        expect(panelContainer.querySelector<HTMLElement>('.aa-Panel')).toBe(
+          root
+        );
+        render(createElement('div', null, 'testSource'), root);
+      },
+    });
+
+    const input = container.querySelector<HTMLInputElement>('.aa-Input');
+
+    fireEvent.input(input, { target: { value: 'a' } });
+
+    await waitFor(() => {
+      expect(
+        panelContainer.querySelector<HTMLElement>('.aa-Panel')
+      ).toBeInTheDocument();
+    });
+  });
+
+  test('provides the state', async () => {
+    const container = document.createElement('div');
+    const panelContainer = document.createElement('div');
+
+    document.body.appendChild(panelContainer);
+    autocomplete<{ label: string }>({
+      container,
+      panelContainer,
+      openOnFocus: true,
+      getSources() {
+        return [
           {
-            sourceId: 'testSource2',
+            sourceId: 'testSource',
+            getItems() {
+              return [{ label: '1' }];
+            },
+            templates: {
+              item({ item }) {
+                return item.label;
+              },
+            },
+          },
+        ];
+      },
+      render({ state, createElement }, root) {
+        expect(state).toEqual({
+          activeItemId: null,
+          collections: [
+            { source: expect.any(Object), items: expect.any(Array) },
+          ],
+          context: expect.any(Object),
+          isOpen: expect.any(Boolean),
+          query: expect.any(String),
+          status: expect.any(String),
+          completion: null,
+        });
+
+        render(createElement('div', null, 'testSource'), root);
+      },
+    });
+
+    const input = container.querySelector<HTMLInputElement>('.aa-Input');
+
+    fireEvent.input(input, { target: { value: 'a' } });
+
+    await waitFor(() => {
+      expect(
+        panelContainer.querySelector<HTMLElement>('.aa-Panel')
+      ).toBeInTheDocument();
+    });
+  });
+
+  test('provides the elements', async () => {
+    const sourceId = 'testSource';
```

Comment:
To keep consistency:

```suggestion
    const sourceId1 = 'testSource1';
```

Reply:
Done in https://github.com/algolia/autocomplete/pull/494/commits/0e2b1f30f70db56d70f5d36bd23aad2d853c7fce